packages/components/src/navigator/navigator-back-button/hook.ts

```diff
@@ -18,23 +19,30 @@ export function useNavigatorBackButton(
 	const {
 		onClick,
 		as = Button,
-		goToParent: goToParentProp = false,
+
+		// Deprecated
+		goToParent,
+
 		...otherProps
 	} = useContextSystem( props, 'NavigatorBackButton' );
 
-	const { goBack, goToParent } = useNavigator();
+	if ( goToParent !== undefined ) {
+		deprecated( '`goToParent` prop in wp.components.NavigatorBackButton', {
+			since: '6.7',
+			alternative:
+				'"back" navigations are always treated as going to the parent screen',
```

Comment:
Should we just suggest `goBack()`? For someone attempting to use `goToParent`, it's not immediately clear in their IDE that `goBack` is the recommended alternative.

Reply:
The `NavigatorBackButton` doesn't have a `goBack` prop, since "going back" was already the default behavior. The `goToParent` prop was changing that behaviour, but now that the default behaviour of "going back" is the same as "going to parent", that prop basically became a no-op and can just be omitted.

Reply:
Ah, got it, makes sense.

Reply:
Actually thanks to your comment, I realised that this deprecation was not necessary because that was a code path that could not be hit anymore. I just removed the code instead: https://github.com/WordPress/gutenberg/pull/63317/commits/da79e357be6f05aa1f02141ad9dd130847c5995f

Reply:
Perfect 🧹